crates/holoom_dna_tests/src/tests/username_registry/external_id_attestation.rs

```diff
@@ -0,0 +1,314 @@
+use hdk::prelude::*;
+use holochain::conductor::api::error::ConductorApiError;
+use holoom_types::{ConfirmExternalIdRequestPayload, ExternalIdAttestation};
+
+use crate::TestSetup;
+
+
+#[tokio::test(flavor = "multi_thread")]
+async fn only_authority_can_create_external_id_attestation() {
+    let setup = TestSetup::authority_and_alice().await;
+
+    // Authority creates a external id Attestation for alice
+    let _: Record = setup
+        .authority_call(
+            "username_registry",
+            "create_external_id_attestation",
+            ExternalIdAttestation {
+                request_id: "1234".into(),
+                internal_pubkey: setup.alice_pubkey(),
+                external_id: "4546".into(),
+                display_name: "alice".into()
+            },
+        )
+        .await
+        .unwrap();
+
+    // Alice cannot create an external id Attestation
+    let result: Result<Record, ConductorApiError> = setup
+        .alice_call(
+            "username_registry",
+            "create_external_id_attestation",
+            ExternalIdAttestation {
+                request_id: "9876".into(),
+                internal_pubkey: setup.alice_pubkey(),
+                external_id: "abcd".into(),
+                display_name: "alice2".into()
+            },
+        )
+        .await;
+
+    assert!(result.is_err());
+}
+
+
+
+//this test anticipates future code changes that chnage tht behaviour 
```

Comment:
Spelling! I know I've asked you delete this test anyway, but this seems like a good opportunity to remind you about:
https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker